packages/typespec-ts/package.json

```diff
@@ -75,20 +75,20 @@
     "@azure/core-util": "^1.4.0",
     "eslint-plugin-require-extensions": "0.1.3",
     "@typespec/ts-http-runtime": "1.0.0-alpha.20240226.9",
-    "@azure-tools/typespec-azure-core": ">=0.39.1 <1.0.0",
+    "@azure-tools/typespec-azure-core": ">=0.40.0 <1.0.0",
     "@azure-tools/typespec-client-generator-core": "0.40.0-dev.17",
```

Comment:
The stable version for 0.40 has issue: https://github.com/Azure/typespec-azure/issues/379, so we would fallback to dev.17 version.